fsd_utils/services/aws.py

```diff
@@ -0,0 +1,337 @@
+import json
+from datetime import datetime
+
+import boto3
+from botocore.exceptions import ClientError
+
+
+class SQSClient:
+    def __init__(
+        self,
+        aws_access_key_id,
+        aws_secret_access_key,
+        region_name="us-west-1",
+        endpoint_url=None,
+        **kwargs,
+    ):
+        self.client = boto3.client(
+            "sqs",
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            region_name=region_name,
+            endpoint_url=endpoint_url,
+            **kwargs,
+        )
+
+    def get_queues(self, prefix=None):
+        """
+        Gets a list of SQS queues. When a prefix is specified, only queues with names
+        that start with the prefix are returned.
+
+        :param prefix: The prefix used to restrict the list of returned queues.
+        :return: A list of Queue names.
+        """
+        if prefix:
+            queues = self.client.list_queues(QueueNamePrefix=prefix)["QueueUrls"]
+        else:
+            queues = self.client.list_queues()["QueueUrls"]
+        if queues:
+            print(f"Got queues: {', '.join([q for q in queues])}")
+            queue_names = [url.split("/")[-1] for url in queues]
+            return queue_names
+        else:
+            print("No queues found.")
+            return []
+
+    def remove_queue(self, queue_url):
+        """
+        Removes an SQS queue. When run against an AWS account, it can take up to
+        60 seconds before the queue is actually deleted.
+
+        :param queue: The queue to delete.
+        :return: None
+        """
+        try:
+            self.client.delete_queue(QueueUrl=queue_url)
+            print(f"Deleted queue with URL={queue_url}.")
+        except ClientError as error:
+            print(f"Couldn't delete queue with URL={queue_url}!")
+            raise error
+
+    def get_queue_url(self, queue_name):
+        response = self.client.get_queue_url(
+            QueueName=queue_name,
+        )
+        return response["QueueUrl"]
+
+    def submit_single_message(
+        self,
+        queue_url,
+        message,
+        extra_attributes: dict = None,
+        message_group_id=None,
+        message_deduplication_id=None,
+    ):
+        try:
+            SQS_CUSTOM_ATTRIBUTES = {
+                "message_created_at": {
+                    "StringValue": str(datetime.now()),
+                    "DataType": "String",
+                },
+            }
+            # add extra message attributes (if provided)
+            if extra_attributes:
+                for key, value in extra_attributes.items():
+                    SQS_CUSTOM_ATTRIBUTES[key] = value
+
+            print(f"Attempting to place message on queue '{queue_url}'.")
+
+            # TODO: Revisit this part after AWS migration
```

Comment:
Do we have an epic we can reference for anything aws migration related? i.e

`# todo(aws-migration)` `# todo(fs-1234)`

Just thinking if we have comments like sentences when searching for them we might miss one or two, if they're annotated differently, so having some unique reference like JIRA id would mitigate that

Reply:
That's a good shout. I'll Check with @aaronwilliamsv1 if there's any epic for this on Jira